classes/form/section_select_form.php

```diff
@@ -100,47 +102,79 @@ public function definition() {
-            // We add the default value: Restore each course module to the section number it has in the source course.
-            $radioarray[] = $mform->createElement('radio', 'targetsectionnum', '',
-            get_string('keepsectionnum', 'block_massaction'), -1, ['class' => 'mt-2']);
-        }
+        $cankeeporiginalsectionnum = ($canaddsection || $srcmaxsectionnum <= $targetsectionnum)
+            && massactionutils::can_keep_original_section_number($targetcourseid, $targetformat->get_format());
+        $this->is_allowed($cankeeporiginalsectionnum, -1, $attributes);
+        $radioarray[] = $mform->createElement('radio', 'targetsectionnum', '',
+            get_string('keepsectionnum', 'block_massaction'), -1, $attributes);
 
         $sectionsrestricted = massactionutils::get_restricted_sections($targetcourseid, $targetformat->get_format());
         // Now add the sections of the target course.
         foreach ($targetsections as $sectionnum => $sectionname) {
-            $attributes = ['class' => 'mt-2'];
-            if (in_array($sectionnum, $sectionsrestricted)) {
-                $attributes['disabled'] = 'disabled';
-            }
+            $this->is_allowed(!in_array($sectionnum, $sectionsrestricted), $sectionnum, $attributes);
             $radioarray[] = $mform->createElement('radio', 'targetsectionnum',
                 '', $sectionname, $sectionnum, $attributes);
         }
 
-        if ($canaddsection) {
-            if (($targetsectionnum + 1) <= $targetformat->get_max_sections()) {
-                // New section option.
-                $radioarray[] = $mform->createElement('radio', 'targetsectionnum', '',
-                    get_string('newsection', 'block_massaction'), $targetsectionnum + 1, ['class' => 'mt-2']);
-            }
-        }
+        // Whether user can add new section.
+        $canaddnewsection = $canaddsection && ($targetsectionnum + 1) <= $targetformat->get_max_sections();
+        $this->is_allowed($canaddnewsection, $targetsectionnum + 1, $attributes);
+        $radioarray[] = $mform->createElement('radio', 'targetsectionnum', '',
+            get_string('newsection', 'block_massaction'), $targetsectionnum + 1, $attributes);
+
 
         $mform->addGroup($radioarray, 'sections', get_string('choosesectiontoduplicateto', 'block_massaction'),
             '<br/>', false);
-        $mform->setDefault('targetsectionnum', -1);
+        $mform->setDefault('targetsectionnum', reset($this->allowedvalues));
 
         $this->add_action_buttons(true, get_string('confirmsectionselect', 'block_massaction'));
     }
+
+    /**
+     * Check if the option should be allowed.
+     * @param bool $allowcondition the condition to allow the option
+     * @param int $optionnumber the option number
+     * @param array $attributes the attributes
+     * @return void
+     */
+    private function is_allowed(bool $allowcondition, int $optionnumber, array &$attributes): void {
```

Comment:
this function name sounds like for checking value and return bool if condition is ok.
maybe `set_radioption` or something?

Reply:
I have changed this to set_radio_option